awx/ui_next/src/screens/Template/shared/SurveyList.jsx

```diff
@@ -1,48 +1,233 @@
+
   const {
-    result: questions,
+    result: { questions, name, description },
     error: contentError,
     isLoading,
     request: fetchSurvey,
   } = useRequest(
     useCallback(async () => {
       const {
-        data: { spec },
+        data: { spec, description: surveyDescription, name: surveyName },
       } = await JobTemplatesAPI.readSurvey(template.id);
-
-      return spec.map((s, index) => ({ ...s, id: index }));
-    }, [template.id])
+      return {
+        questions: spec.map((s, index) => ({ ...s, id: index })),
+        description: surveyDescription,
+        name: surveyName,
+      };
+    }, [template.id]),
+    { questions: [], name: '', description: '' }
   );
 
   useEffect(() => {
     fetchSurvey();
   }, [fetchSurvey]);
-  if (contentError) {
+
+  const isAllSelected =
+    selected.length === questions?.length && selected.length > 0;
+
+  const handleSelectAll = isSelected => {
+    setSelected(isSelected ? [...questions] : []);
+  };
+
+  const handleSelect = item => {
+    if (selected.some(s => s.id === item.id)) {
+      setSelected(selected.filter(s => s.id !== item.id));
+    } else {
+      setSelected(selected.concat(item));
+    }
+  };
+
+  const {
+    isLoading: isDeleteLoading,
+    deleteItems: deleteQuestions,
+    deletionError,
+    clearDeletionError,
+  } = useDeleteItems(
+    useCallback(async () => {
+      if (isAllSelected) {
+        return JobTemplatesAPI.destroySurvey(template.id);
+      }
+      const surveyQuestions = [];
+      questions.forEach(q =>
+        !selected.find(s => s.id === q.id) ? surveyQuestions.push(q) : null
+      );
+      return JobTemplatesAPI.updateSurvey(template.id, {
+        name,
+        description,
+        spec: surveyQuestions,
+      });
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selected]),
+    {
+      qsConfig: QS_CONFIG,
+      fetchItems: fetchSurvey,
+    }
+  );
+
+  const handleDelete = async () => {
+    await deleteQuestions();
+    setIsDeleteModalOpen(false);
+    setSelected([]);
+  };
+
+  const { isToggleLoading, error, request: toggleSurvey } = useRequest(
+    useCallback(async () => {
+      await JobTemplatesAPI.update(template.id, {
+        survey_enabled: !surveyEnabled,
+      });
+      return setSurveyEnabled(!surveyEnabled);
+    }, [template, surveyEnabled]),
+    template.survey_enabled
+  );
+
+  if (contentError || error) {
     return <ContentError error={contentError} />;
   }
-  if (isLoading) {
+
+  if (isLoading || isToggleLoading || isDeleteLoading) {
     return <ContentLoading />;
   }
 
+  if (deletionError) {
+    return (
+      <AlertModal
+        isOpen={deletionError}
+        variant="error"
+        title={i18n._(t`Error!`)}
+        onClose={clearDeletionError}
+      >
+        {i18n._(t`Failed to delete one or more jobs.`)}
+        <ErrorDetail error={deletionError} />
+      </AlertModal>
+    );
+  }
+
+  if (isDeleteModalOpen) {
+    return (
```

Comment:
if I'm understanding this right, this early return will skip rendering the entire survey list, panel, etc. Is that the expected behavior? Generally with modals we render them on top of the existing UI.